test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -222,3 +222,68 @@ timestamp_manager::validate_read_timestamp(session_simulator *session, const uin
 
     return (0);
 }
+
+/* Validate the commit timestamp. */
+int
+timestamp_manager::validate_commit_timestamp(
+  session_simulator *session, const uint64_t commit_ts) const
+{
+    uint64_t prepare_ts = session->get_prepare_timestamp();
+    if (!prepare_ts) {
+        /*
+         * We cannot set the commit timestamp to be earlier than the fist commit timestamp when
+         * setting the commit timestamp multiple times within a transaction.
+         */
+        uint64_t first_commit_ts = session->get_first_commit_timestamp();
```

Comment:
The first_commit_ts is not reset for transaction, I will work on it as part of WT-9242